exonum-java-binding-core/src/test/java/com/exonum/binding/storage/indices/ProofListContainsMatcher.java

```diff
@@ -69,38 +69,33 @@ public void describeTo(Description description) {
   @Override
   protected void describeMismatchSafely(ProofListIndexProxy<String> list,
                                         Description mismatchDescription) {
-    ListProof proof = proofFunction.apply(list);
+    ListProofNode proof = proofFunction.apply(list);
 
-    ListProofStructureValidator validator = newProofStructureValidator(proof);
-    ListProofRootHashCalculator<String> calculator = newProofHashCodeCalculator(proof);
+    CheckedListProof checkedProof = newCheckedListProof(proof);
 
-    if (!validator.isValid()) {
+    if (!checkedProof.isValid()) {
       mismatchDescription.appendText("proof was not valid: ")
-          .appendValue(validator.getProofStatus().getDescription());
+          .appendValue(checkedProof.getProofStatus().getDescription());
       return;
     }
 
-    if (!list.getRootHash().equals(calculator.getCalculatedRootHash())) {
+    if (!list.getRootHash().equals(checkedProof.getRootHash())) {
       mismatchDescription.appendText("calculated root hash doesn't match: ")
-          .appendValue(calculator.getCalculatedRootHash())
+          .appendValue(checkedProof.getRootHash())
           .appendText("expected root hash: ")
           .appendValue(list.getRootHash());
       return;
     }
 
-    if (!elementsMatcher.matches(calculator.getElements())) {
-      mismatchDescription.appendText("valid proof: ").appendValue(validator)
+    if (!elementsMatcher.matches(checkedProof.getElements())) {
+      mismatchDescription.appendText("valid proof: ").appendValue(checkedProof)
           .appendText(", elements mismatch: ");
-      elementsMatcher.describeMismatch(calculator.getElements(), mismatchDescription);
+      elementsMatcher.describeMismatch(checkedProof.getElements(), mismatchDescription);
     }
   }
 
-  private ListProofStructureValidator newProofStructureValidator(ListProof listProof) {
-    return new ListProofStructureValidator(listProof);
-  }
-
-  private ListProofRootHashCalculator<String> newProofHashCodeCalculator(ListProof listProof) {
-    return new ListProofRootHashCalculator<>(listProof, StandardSerializers.string());
+  private CheckedListProof newCheckedListProof(ListProofNode listProof) {
```

Comment:
This name is misleading: we check a proof, not create one.
However, shan't ProofList#getProof return a ListProofNode, and not an UncheckedListProof, as discussed?